jib-build-plan/src/main/java/com/google/cloud/tools/jib/api/buildplan/FilePermissionsProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.api.buildplan;
+
+import java.nio.file.Path;
+
+/** Serves as a base class for determining file permissions on a container. */
```

Comment:
Remove "Servies as a base class" as an interface is not really a "base class"

Maybe it could be something like "Interface for providing rules to determine file permission on a container"

Reply:
and similarly for the other interfaces?

Reply:
That is much better, thanks.